From 27ecf4da5ad3845b773508917e71f5a07b149077 Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Thu, 2 Jul 2026 01:45:40 +0000
Subject: [PATCH 03843/15333] ASoC: simple-card: tidyup simple_util_parse_xxx()
 in simple_parse_of()

simple_parse_of() calls simple_util_parse_xxx(), but are random.
Let's gather them all in one place.

Let's keep same style with audio-graph-card/audio-graph-card2.

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Link: https://patch.msgid.link/877bnexiyj.wl-kuninori.morimoto.gx@renesas.com
Signed-off-by: Mark Brown <broonie@kernel.org>
---
 sound/soc/generic/simple-card.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/sound/soc/generic/simple-card.c b/sound/soc/generic/simple-card.c
index 4dd05e726213..a67f4aad0188 100644
--- a/sound/soc/generic/simple-card.c
+++ b/sound/soc/generic/simple-card.c
@@ -704,23 +704,23 @@ static int simple_parse_of(struct simple_util_priv *priv)
 	if (ret < 0)
 		goto end;
 
-	/* Single/Muti DAI link(s) & New style of DT node */
-	memset(li, 0, sizeof(*li));
-	ret = simple_for_each_link(priv, li,
-				   simple_dai_link_of,
-				   simple_dai_link_of_dpcm);
+	ret = simple_util_parse_card_name(priv, PREFIX);
 	if (ret < 0)
 		goto err;
 
-	ret = simple_util_parse_card_name(priv, PREFIX);
+	ret = simple_util_parse_aux_devs(priv, PREFIX);
 	if (ret < 0)
 		goto err;
 
-	ret = simple_populate_aux(priv);
+	/* Single/Muti DAI link(s) & New style of DT node */
+	memset(li, 0, sizeof(*li));
+	ret = simple_for_each_link(priv, li,
+				   simple_dai_link_of,
+				   simple_dai_link_of_dpcm);
 	if (ret < 0)
 		goto err;
 
-	ret = simple_util_parse_aux_devs(priv, PREFIX);
+	ret = simple_populate_aux(priv);
 	if (ret < 0)
 		goto err;
 
-- 
2.55.0

